frank-flow/src/frontend/src/app/editor/explorer/add-dialog/add-dialog.component.ts

```diff
@@ -28,8 +28,62 @@ export class AddDialogComponent {
     this.currentDirectory = this.codeService.currentDirectory;
   }
 
+  findDuplicateFileName(fileName: string, directory: any): boolean {
```

Comment:
We should create a model for the file-structure.

Reply:
I will make a feature issue and add it to things to refactor.